langlib/langlib-test/src/test/resources/test-src/functionlib_test.bal

```diff
@@ -228,28 +228,28 @@ function testCallFunctionWithInvalidArguments() {
     a1 = trap function:call(value8, "");
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[int...]', found '[string]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: expected argument(s) of type(s) 'int...', found 'string'", (<error> a1).detail()["message"]);
 
     a1 = trap function:call(value8, "", "");
     assertEquality(true, a1 is error);
     assertEquality("{ballerina/lang.function}IncompatibleArguments", (<error> a1).message());
-    assertEquality("arguments of incompatible types: expected argument(s) of types '[int...]', found '[string,string]'", (<error> a1).detail()["message"]);
+    assertEquality("arguments of incompatible types: expected argument(s) of type(s) 'int...', found 'string,string'", (<error> a1).detail()["message"]);
```

Comment:
Sorry, missed this earlier, but with rest params this message doesn't quite make sense does it? 

Can it be something like

```
arguments of incompatible types: argument(s) of type(s) 'string,string' cannot be passed to function with signature '(int...)'
```

I guess we lose defaultable param info here, but mismatched argument counts are handled separately anyway, right?

Reply:
So if we are going to mention it as function signature then we need to add the return type of that function to the error message also.

Reply:
Maybe

```
arguments of incompatible types: argument(s) of type(s) 'string,string' cannot be passed to function expecting parameters '(int...)'
```

Reply:
Change the error message as
```
arguments of incompatible types: argument(s) of type(s) 'string,string' cannot be passed to function expecting parameter(s) of type(s)  'int...'
```